southbound/impl/src/main/java/org/opendaylight/ansible/southbound/StatusL3vpnListener.java

```diff
@@ -0,0 +1,242 @@
+        // Kick off ansible processes in parallel, then use Futures to callback when it is all done
+
+        List<ListenableFuture<Uuid>> cmdFutures = new ArrayList<>();
+        Sites sites = l3vpnSvc.getSites();
+        if (sites == null) {
+            LOG.error("L3 VPN config missing sites configuration");
+            throw new L3vpnConfigException("L3 VPN config missing sites");
+        }
+        List<Site> siteList = sites.getSite();
+        if (siteList == null || siteList.isEmpty()) {
+            LOG.error("L3 VPN config missing sites configuration");
+            throw new L3vpnConfigException("L3 VPN config missing site configuration");
+        }
+
+        for (Site site: siteList) {
+            if (site == null) {
+                LOG.warn("Null site found in L3 VPN config...ignoring");
+                continue;
+            }
+            LOG.info("Examining site " + site.toString());
+            cmdFutures.add(configureSite(site));
+        }
+
+        l3vpnCommands = Futures.successfulAsList(cmdFutures);
+        l3vpnCommands.addListener(() -> {
+            try {
+                l3vpnCommands.get().forEach(uuid -> {
+                    if (uuid == null) {
+                        txRunner.callWithNewReadWriteTransactionAndSubmit(OPERATIONAL, tx ->
+                            IetfL3vpnSvcUtils.putStatusL3VPN(tx, COMMIT_VERSION, Status.Failed)
+                        );
+                        return;
+                    }
+                });
+                txRunner.callWithNewReadWriteTransactionAndSubmit(OPERATIONAL, tx ->
+                    IetfL3vpnSvcUtils.putStatusL3VPN(tx, COMMIT_VERSION, Status.Complete)
+                );
+            } catch (InterruptedException | ExecutionException e) {
+                txRunner.callWithNewReadWriteTransactionAndSubmit(OPERATIONAL, tx ->
+                    IetfL3vpnSvcUtils.putStatusL3VPN(tx, COMMIT_VERSION, Status.Failed)
+                );
+            }
+        }, executor);
+        return Status.InProgress;
+    }
+
+    private ListenableFuture<Uuid> configureSite(Site site) throws AnsibleCommandException, L3vpnConfigException {
+        // Find management info for Ansible
+        final PeAugmentation siteAccess;
+        List<SiteNetworkAccess> access = site.getSiteNetworkAccesses().getSiteNetworkAccess();
+        final List<String> roleVars;
+        final List<String> ansibleVars;
+        final String deviceType;
+        final String provider;
+        final String networkOS;
+        final Uuid siteCmdId;
+
+        LOG.info("Configuring site" + site.toString());
+        if (access == null || access.isEmpty()) {
+            LOG.error("Site Network Access information is missing for site {}", site);
+            throw new L3vpnConfigException("Missing site information");
+        }
+        // Assume only 1 access
+        LOG.info("Site Network Access list is: " + access.toString());
```

Comment:
toString is not needed and prints the whole class, what you really want is getValue() to get just the string value.

Reply:
done

Reply:
acces doesnt have a getValue method

Reply:
oh this is a list and not the final class.